### DISCUSSION TITLE
Element are closing on click to dropdown

### DISCUSSION THREAD

**Discussion by bubnenkoff** [alpinejs/alpine#2484] [1. Help]
I am trying to create simple dropdown with input fields. The problem that it's closing every time when I am clicking on input field. How to fix it?

https://codepen.io/bubnenkoff/pen/MWEjrYO

And the second question. Is it possible to use something another except button to create drop-down list? I tried to change `button` to some another html tags. It's stopped works . I want to create drop-down menu with items.

**Comment by SimoTod** [comment]
Don't put your dropdown inside the button otherwise when you click on the dropdown, the click event will bubble up to the button tag and will set the property to false again.

**Comment by bubnenkoff** [comment]
What tags I should use instead?


Пт, 10 дек. 2021 г. в 21:52, Simone Todaro ***@***.***>:

**Comment by bubnenkoff** [comment]
Could you show an example how code should be organized?

Пт, 10 дек. 2021 г. в 21:58, Dmitry B ***@***.***>:

**Reply by SimoTod** [reply]
You have a few options, here some of them: https://codepen.io/SimoTod/pen/abLmEro (I didn't put any styles exept for an example showing how to change the arrows for a detail tag)

Example 1. Browsers already support it, if you don't need complex logic or animation, you can do it without Alpine.
Example 2. This is how I would build a dropdown in Alpine.
Example 3. It's your original example but fixed. Basically, I wrapped the dropdown content in an additional div with @click.stop that would prevent the click event to reach the button tag when you click inside it.
